hexrd/instrument.py

```diff
@@ -3672,18 +3674,66 @@ def __init__(self, image_series_dict, instrument, plane_data,
             self._iHKLList = active_hkls
             n_rings = len(active_hkls)
 
+        # grab a det key and corresponding imageseries (first will do)
+        # !!! assuming that the imageseries for all panels
+        #     have the same length and omegas
+        det_key = list(eta_mapping.keys())[0]
```

Comment:
The `eta_mapping` variable gets created via `instrument.extract_polar_maps()` below. Maybe you need to move that call to before this line?

Reply:
yep.